### DISCUSSION TITLE
Restoring Bitwarden DB

### DISCUSSION THREAD

**Discussion by sebasptsch** [hassio-addons/addon-bitwarden#137] [Q&A] [answered]
I'm migrating my bitwarden data from a seperate docker host to home assistant's bitwarden addon. How can I access or modify the addon's data?

I've attempted to extract a snapshot and replace it's files but can't seem to get the directory structure right.

**Comment by muxa** [comment]
While searching for issues related to this topic I came across https://github.com/hassio-addons/addon-bitwarden/issues/32 where @frenck said:

> We can't expose the data directory.

@frenck can you please explain what the limitations are? Ideally I want to migrate by Bitwarden DB from a separate docker container to the Add On.

**Reply by muxa** [reply]
I turns out Bitwarden has Export/Import feature under Tools in the Web UI. I just migrated my data using that.

**Reply by frenck** [reply]
> can you please explain what the limitations are?

Sure. There is simply no such option... Not sure what else you want me to say 🤷 
There is an RFC open to adding such capability to the Supervisor.

**Comment by sebasptsch** [comment, marked as answer]
For people just arriving to this thread, I was able to do something quite unorthodox where I enabled ssh into port 2222 and copied the data this way. Since then it's been working fine.

At the moment there is really no other way to do this besides a per-user approach mentioned by @muxa.
